### PR TITLE
Add stop_machine for action :stop, fixes #162

### LINKED ISSUE #162: Would be nice to have :stop action on machines
Currently, it will fail with:
```
 RuntimeError: Chef::Provisioning::AWSDriver::Driver does not implement stop_machine
```

### PR DESCRIPTION
**Pull request by jkeiser**
fixes https://github.com/chef/chef-provisioning-aws/issues/162
fixes https://github.com/chef/chef-provisioning-aws/issues/370

### CODE REVIEW

**Comment by tyler-ball** [comment]
@jkeiser Is this still WIP?

**Comment by jkeiser** [comment]
@tyler-ball It looks pretty sane to me, and yeah, it needs tests.

**Comment by tyler-ball** [comment]
I'm :+1: on the code as it is - I think it only needs a single test and a rebase to get it merged.

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 1]
```diff
@@ -464,6 +464,19 @@ def connect_to_machine(name, chef_server = nil)
       machine_for(machine_spec, machine_spec.reference)
     end
 
+    def stop_machine(action_handler, machine_spec, machine_options)
+      instance = instance_for(machine_spec)
+      if instance && instance.exists?
```
Comment (randomcamel) [inline lib/chef/provisioning/aws_driver/driver.rb]: nnnnggh `exists?`

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 2]
```diff
@@ -464,6 +464,19 @@ def connect_to_machine(name, chef_server = nil)
...
+        wait_until_machine(action_handler, machine_spec, "finish coming up so we can stop it", "up", instance) { instance.status != :pending }
+        if instance.status == :running
+          action_handler.perform_action "Stop #{machine_spec.name} (#{instance.id}) in #{aws_config.region} ..." do
+            instance.stop
+          end
+        end
+        wait_until_machine(action_handler, machine_spec, "stop", "stopped", instance) { [ :stopped, :terminated ].include?(instance.status) }
```
Comment (randomcamel) [inline lib/chef/provisioning/aws_driver/driver.rb]: Everyone learns the same lesson the hard way about `stop`ping an ephemeral instance...

**Comment by randomcamel** [comment]
:+1: with rebase + test.

File: lib/chef/provisioning/aws_driver/driver.rb (hunk 1 above)
Comment (tyler-ball) [inline lib/chef/provisioning/aws_driver/driver.rb]: What does the 'nnnnggh' mean?

Comment (randomcamel) [inline lib/chef/provisioning/aws_driver/driver.rb]: We'd had some conversations elsewhere about the idea of "exists" having a variety of meanings where AWS instances are concerned, and using a different set of query functions to determine the exact state needed.

**Pull request merged**